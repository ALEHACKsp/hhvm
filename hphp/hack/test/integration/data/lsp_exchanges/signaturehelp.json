[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializationOptions": {},
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "$test/waitForHhServerReady",
    "params": {}
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}",
        "languageId": "hack",
        "version": 1,
        "text": "${php_file}"
      }
    }
  },
  {
    "comment": "signature help for 0-argument constructor (left of opening paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 10,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 16,
        "character": 18
      }
    }
  },
  {
    "comment": "signature help for 0-argument constructor",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 20,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 16,
        "character": 19
      }
    }
  },
  {
    "comment": "signature help for 0-argument constructor (right of closing paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 30,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 16,
        "character": 20
      }
    }
  },
  {
    "comment": "signature help for 2-argument instance method (left of opening paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 40,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 17,
        "character": 20
      }
    }
  },
  {
    "comment": "signature help for 2-argument instance method (right of opening paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 50,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 17,
        "character": 21
      }
    }
  },
  {
    "comment": "signature help for 2-argument instance method (left of first comma)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 60,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 17,
        "character": 22
      }
    }
  },
  {
    "comment": "signature help for 2-argument instance method (right of first comma)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 70,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 17,
        "character": 23
      }
    }
  },
  {
    "comment": "signature help for 2-argument instance method (left of closing paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 80,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 17,
        "character": 24
      }
    }
  },
  {
    "comment": "signature help for 2-argument instance method (right of closing paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 90,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 17,
        "character": 25
      }
    }
  },
  {
    "comment": "signature help for 1-argument static method (left of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 100,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 18,
        "character": 23
      }
    }
  },
  {
    "comment": "signature help for 1-argument static method (right of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 110,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 18,
        "character": 24
      }
    }
  },
  {
    "comment": "signature help for 2-argument global function (left of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 120,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 19,
        "character": 17
      }
    }
  },
  {
    "comment": "signature help for 2-argument global function (right of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 130,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 19,
        "character": 18
      }
    }
  },
  {
    "comment": "signature help for 1-argument namespace-aliased global function (left of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 140,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 20,
        "character": 26
      }
    }
  },
  {
    "comment": "signature help for 1-argument namespace-aliased global function (right of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 150,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 20,
        "character": 26
      }
    }
  },
  {
    "comment": "signature help for 1-argument namespace-aliased global function (right of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 160,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 20,
        "character": 27
      }
    }
  },
  {
    "comment": "signature help for 1-argument namespace-aliased global function (right of open paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 170,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 20,
        "character": 28
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  }
]
