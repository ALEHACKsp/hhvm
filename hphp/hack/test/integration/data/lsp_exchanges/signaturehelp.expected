[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "typeDefinitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 10,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 20,
    "result": {
      "signatures": [
        {
          "label": "public function __construct(): _",
          "documentation": "Constructor with doc block",
          "parameters": []
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 30,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 40,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 50,
    "result": {
      "signatures": [
        {
          "label": "public function instanceMethod(int $x1, int $x2): void",
          "documentation": "Instance method with doc block",
          "parameters": [
            {
              "label": "$x1"
            },
            {
              "label": "$x2"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 60,
    "result": {
      "signatures": [
        {
          "label": "public function instanceMethod(int $x1, int $x2): void",
          "documentation": "Instance method with doc block",
          "parameters": [
            {
              "label": "$x1"
            },
            {
              "label": "$x2"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 1
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 70,
    "result": {
      "signatures": [
        {
          "label": "public function instanceMethod(int $x1, int $x2): void",
          "documentation": "Instance method with doc block",
          "parameters": [
            {
              "label": "$x1"
            },
            {
              "label": "$x2"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 1
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 80,
    "result": {
      "signatures": [
        {
          "label": "public function instanceMethod(int $x1, int $x2): void",
          "documentation": "Instance method with doc block",
          "parameters": [
            {
              "label": "$x1"
            },
            {
              "label": "$x2"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 1
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 90,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 100,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 110,
    "result": {
      "signatures": [
        {
          "label": "public static function staticMethod(string $z): void",
          "documentation": "Static method with doc block",
          "parameters": [
            {
              "label": "$z"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 120,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 130,
    "result": {
      "signatures": [
        {
          "label": "function global_function(string $s, int $x): void",
          "documentation": "Global function with doc block",
          "parameters": [
            {
              "label": "$s"
            },
            {
              "label": "$x"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 140,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 150,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 160,
    "result": {
      "signatures": [
        {
          "label": "function Herp\\aliased_global_func(string $s): void",
          "parameters": [
            {
              "label": "$s"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 170,
    "result": {
      "signatures": [
        {
          "label": "function Herp\\aliased_global_func(string $s): void",
          "parameters": [
            {
              "label": "$s"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]
